File: folioreader/src/main/java/com/folioreader/ui/base/FolioBookHolder.java
package com.folioreader.ui.base;

import com.folioreader.model.HighlightImpl;
import com.folioreader.model.search.SearchItem;

public interface FolioBookHolder {
    String getCurrentHref();
    void highlight(HighlightImpl.HighlightStyle style, boolean isAlreadyCreated);
    String getPageName();
    void loadRangy(String rangy);
    void setSearchItemVisible(SearchItem item);
}

File: folioreader/src/main/java/com/folioreader/ui/base/FolioBookHolder.kt
package com.folioreader.ui.base

import android.graphics.Rect
import com.folioreader.model.HighlightImpl
import com.folioreader.model.locators.SearchLocator

interface FolioBookHolder {
    val currentHref: String?
    val pageName: String
    fun highlight(style: HighlightImpl.HighlightStyle, isAlreadyCreated: Boolean)
    fun loadRangy(rangy: String)
    fun loadMarker(rangy: String, globalIds: String)
    var searchLocatorVisible: SearchLocator?
    fun showMenu(): Boolean {
        return true
    }
    fun triggerHighlight(rect: Rect) {
        print("triggerHighlight at $rect)")
    }
}
